homeassistant/components/swiss_public_transport/sensor.py

```diff
@@ -107,47 +140,54 @@ class SwissPublicTransportSensor(
 ):
     """Implementation of a Swiss public transport sensor."""
 
+    entity_description: SwissPublicTransportSensorEntityDescription
     _attr_attribution = "Data provided by transport.opendata.ch"
-    _attr_icon = "mdi:bus"
-    _attr_has_entity_name = True
-    _attr_translation_key = "departure"
-    _attr_device_class = SensorDeviceClass.TIMESTAMP
+
+    _native_key: str
 
     def __init__(
         self,
         coordinator: SwissPublicTransportDataUpdateCoordinator,
+        entity_description: SwissPublicTransportSensorEntityDescription,
         unique_id: str,
     ) -> None:
         """Initialize the sensor."""
         super().__init__(coordinator)
-        self._attr_unique_id = f"{unique_id}_departure"
+        self.entity_description = entity_description
+        self._attr_unique_id = f"{unique_id}_{self.entity_description.key}"
         self._attr_device_info = DeviceInfo(
             identifiers={(DOMAIN, unique_id)},
             manufacturer="Opendata.ch",
             entry_type=DeviceEntryType.SERVICE,
         )
 
+    @property
+    def native_value(self) -> datetime | None:
+        """Return the state of the sensor."""
+        return self.entity_description.value_fn(
+            self.coordinator.data[self.entity_description.index]
+        )
+
     async def async_added_to_hass(self) -> None:
         """Prepare the extra attributes at start."""
-        self._async_update_attrs()
+        if self.entity_description.has_legacy_attributes:
+            self._async_update_attrs()
```

Comment:
Unused?

Reply:
No, the first connections still contains ALL the extra attributes. With this request, I only broke out the timestamps, the rest will come later. Remove it now will result in loss of functionality.

Reply:
I know, but I haven't seen any entity declared that uses this flag (maybe I'm missing something, so please forgive me if I'm overlooking)

Reply:
Good spot, that got lost in one of the commits. I defined it again so only the first sensor stores the additional information.